### PR TITLE
Integrate Firestore emulator

### PR DESCRIPTION
**Pull request by eweitz**
This integrates the [Firestore emulator](https://cloud.google.com/sdk/gcloud/reference/beta/emulators/firestore/) into automated tests for Ingest Pipeline.  Benefits include:

* Faster development.  Rather than writing to a remote Firestore, engineers can write to a local Firestore.  Manual tests indicate a roughly 4x speedup (~30 seconds before, ~7 seconds after) for our [test matrix](https://github.com/broadinstitute/scp-ingest-pipeline/commit/bbfa5d4d3e6e0afa6d248201eeecee8d40238758#diff-b77fbe45bbe85acb9b32cc7a810d777b).

* More code coverage.  Before, we could not quickly test the "load" phase of this ETL pipeline.  Now we do.  Compare coverage [before](https://249-187902423-gh.circle-artifacts.com/0/home/circleci/scp-ingest-pipeline/tests/htmlcov/_home_circleci_scp-ingest-pipeline_ingest_ingest_pipeline_py.html#n108) and [after](https://258-187902423-gh.circle-artifacts.com/0/home/circleci/scp-ingest-pipeline/tests/htmlcov/_home_circleci_scp-ingest-pipeline_ingest_ingest_pipeline_py.html#n112).  This enables fast, reliable integration tests.

To set up Firestore emulator locally, ensure you have Java (JRE >=8) installed, then:
* `gcloud components install beta cloud-firestore-emulator`
* `gcloud beta emulators firestore start --host-port localhost:8081`
* In another Terminal:
`export FIRESTORE_EMULATOR_HOST=localhost:8081`
* Run your development command, e.g.:
`python3 ingest_pipeline.py  --study-accession SCP1 --file-id 123abc ingest_expression --taxon-name 'Homo sapiens' --taxon-common-name human --ncbi-taxid 9606 --matrix-file ../tests/data/dense_matrix_19_genes_100k_cells.txt --matrix-file-type dense`

That approach works for Ingest Pipeline development.

Development on the Rails codebase could theoretically also then point to this local Firestore, assuming the file metadata being kept in MongoDB were already populated.  The approach described in the [`google-cloud-ruby`](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-firestore/EMULATOR.md#google-cloud-firestore-emulator) library is probably the best starting point, if we pursued that.  The steps above for "set up Firestore locally" could likely be used in place of the Firebase-specific instructions in that `google-cloud-ruby` link.

This satisfies [SCP-1830](https://broadworkbench.atlassian.net/browse/SCP-1830).

### CODE REVIEW

File: tests/test_ingest.py [hunk 1]
```diff
@@ -31,74 +31,73 @@
...
-    Use Google's official Firestore mock,
-    https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-firestore/EMULATOR.md#google-cloud-firestore-emulator
 
-    This will enable us to also verify (and thus cover) loading code *outputs*,
-    unlike here where we merely give a way to verify loading code *inputs*.
-    Doing so via integration tests will isolate us from implementation changes.
-    """
-    self.load_expression_data_args = args
-    self.load_expression_data_kwargs = kwargs
+def get_random_study_accession():
```
Comment (knapii-developments) [inline tests/test_ingest.py]: Nice work around for FireStore!

File: tests/test_ingest.py [hunk 2]
```diff
@@ -31,74 +31,73 @@
...
+    docs = sorted(docs, key=lambda doc: doc['name'])
+
+    actual_doc = docs[n]
 
     # Uncomment to print out new baseline data
     # Process to update baselines is manual: copy and paste it into new file
     # TODO: Automate when reasonable
-    print("actual_model")
-    print(actual_model)
+    print("actual_doc")
```
Comment (knapii-developments) [inline tests/test_ingest.py]: You could use an F-string here. But up to you.

File: tests/test_ingest.py [hunk 3]
```diff
@@ -181,9 +190,11 @@ def test_ingest_local_dense_matrix(self):
         """Ingest Pipeline should extract and transform local dense matrices
         """
 
+        study_accession = get_random_study_accession()
+
         args = [
```
Comment (knapii-developments) [inline tests/test_ingest.py]: It looks like you have 2 set of args here. Is that to be as aspected?

File: tests/test_ingest.py (hunk 2 above)
Comment (eweitz) [reply tests/test_ingest.py]: [Done!](https://github.com/broadinstitute/scp-ingest-pipeline/pull/26/commits/a766165a9823cc67f287e8ea26d6ec255c8fe09e#diff-9398b55cff89e452e5b4049aceaac50cR71)

File: tests/test_ingest.py (hunk 1 above)
Comment (eweitz) [reply tests/test_ingest.py]: Thanks go to @bistline for that idea.

File: tests/test_ingest.py [hunk 4]
```diff
@@ -204,26 +215,27 @@ def test_ingest_local_dense_matrix(self):
...
+        num_docs = len(docs)
+        self.assertEqual(num_docs, 19)
 
     def test_ingest_missing_local_file(self):
         """Ingest Pipeline should throw error for missing local file
         """
 
+        study_accession = get_random_study_accession()
+
         args = [
```
Comment (knapii-developments) [inline tests/test_ingest.py]: You have double args here as well.

File: tests/test_ingest.py (hunk 3 above)
Comment (eweitz) [reply tests/test_ingest.py]: Per our chat, there is actually only one set of arguments for each test, but the GitHub diff UI makes it look like two.

File: tests/test_ingest.py (hunk 4 above)
Comment (eweitz) [reply tests/test_ingest.py]: This is the diff UI issue [described above](https://github.com/broadinstitute/scp-ingest-pipeline/pull/26#discussion_r325794668); there's actual just one set of args.

**Pull request merged**